packages/libs/eda/src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -745,8 +748,14 @@ function ScatterplotViz(props: VisualizationProps<Options>) {
         config: {
           outputEntityId: outputEntity.id,
           valueSpec: hideTrendlines ? undefined : valueSpecValue,
-          xAxisVariable: vizConfig.xAxisVariable,
-          yAxisVariable: vizConfig.yAxisVariable,
+          xAxisVariable:
+            sendComputedVariablesInRequest && computedXAxisDescriptor
+              ? computedXAxisDescriptor
+              : vizConfig.xAxisVariable,
```

Comment:
Line 382 (`variablesForConstraints`) should probably be calculated consistently too? (including `sendComputedVariablesInRequest` in the conditional?

e.g.
```
  xAxisVariable: sendComputedVariablesInRequest && computedXAxisDescriptor
    ? computedXAxisDescriptor
    : vizConfig.xAxisVariable,
```